core/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -426,6 +436,50 @@ public MetadataResourceContainer getResourceContainerDescription(final ServiceCo
         }
     }
 
+    @Override
+    public void copyResources(ServiceContext context, String sourceUuid, String targetUuid, MetadataResourceVisibility metadataResourceVisibility, boolean sourceApproved, boolean targetApproved) throws Exception {
+        final int sourceMetadataId = canEdit(context, sourceUuid, metadataResourceVisibility, sourceApproved);
+        final String sourceResourceTypeDir = getMetadataDir(context, sourceMetadataId) + cmisConfiguration.getFolderDelimiter() + metadataResourceVisibility.toString();
+        try {
+            Folder sourceParentFolder = (Folder) cmisConfiguration.getClient().getObjectByPath(sourceResourceTypeDir);
+
+            Map<String, Document> sourceDocumentMap = cmisUtils.getCmisObjectMap(sourceParentFolder, null);
+
+
+            for (Map.Entry<String, Document> sourceEntry : sourceDocumentMap.entrySet()) {
+                Document sourceDocument = sourceEntry.getValue();
+                if (sourceDocument instanceof Document) {
+                    // Get cmis properties from the source document
+                    Map<String, Object> sourceProperties = getSecondaryProperties(sourceDocument);
+                    putResource(context, targetUuid, sourceDocument.getName(), sourceDocument.getContentStream().getStream(), null, metadataResourceVisibility, targetApproved, sourceProperties);
```

Comment:
There are regular CMIS properties and there are secondary properties.
You are passing the sourceProperties to the put resource function which are the secondary properties but you don't seem to be also including the regular properties in the list as well.

i.e. if the cmis:description property were set then it would get lots in the copy.

I believe the Document.copy command does copy the properties.

So you could do something along these lines.

```
                Document sourceDocument = sourceEntry.getValue();
                final int metadataId = canEdit(context, targetUuid, targetApproved);
                String destKey = getKey(context, targetUuid, metadataId, metadataResourceVisibility, sourceDocument.getName());
                int lastFolderDelimiterDestKeyIndex = destKey.lastIndexOf(cmisConfiguration.getFolderDelimiter());
                String parentDestFolderKey = destKey.substring(0, lastFolderDelimiterDestKeyIndex);
                Folder parentDestFolder = cmisUtils.getFolderCache(parentDestFolderKey, true);
                Document destDocument = sourceDocument.copy(parentDestFolder);
                setCmisMetadataUUIDSecondary(destDocument, getSecondaryProperties(sourceDocument), targetUuid);
```

Or an alternative could be to modify the sourceProperties so that it include both primary and secondary properties.

Reply:
I tried. The backend content server is not happy with overwriting all properties. I can only do these secondary properties without causing crashing.

Reply:
I did some test and you are correct - the copy complains that document already exists.
I guess you can leave it as is for now until we identify exactly what properties would need to be copied